components-core/src/main/java/com/airwallex/android/core/log/AnalyticsLogger.kt

```diff
@@ -0,0 +1,97 @@
+package com.airwallex.android.core.log
+
+import android.content.Context
+import com.airwallex.airtracker.Config
+import com.airwallex.airtracker.Environment
+import com.airwallex.airtracker.Tracker
+import com.airwallex.android.core.AirwallexPlugins
+import com.airwallex.android.core.BuildConfig
+import com.airwallex.android.core.TokenManager
+import com.airwallex.android.core.exception.AirwallexException
+import com.airwallex.android.core.extension.getAppName
+import com.airwallex.android.core.extension.getAppVersion
+import com.airwallex.android.core.extension.putIfNotNull
+
+private typealias AirwallexEnviornment = com.airwallex.android.core.Environment
+
+object AnalyticsLogger {
+    private var tracker: Tracker? = null
+
+    fun initialize(context: Context) {
+        if (AirwallexPlugins.enableAnalytics && tracker == null) {
+            tracker = Tracker(
+                Config(
+                    appName = "pa_mobile_sdk",
+                    appVersion = BuildConfig.VERSION_NAME,
+                    environment = AirwallexPlugins.environment.toTrackerEnvironment()
+                ),
+                context
+            )
+            tracker?.extraCommonData = getExtraCommonData(context)
+        }
+    }
+
+    fun updateAccountId(accountId: String?) {
+        tracker?.let {
+            it.extraCommonData = it.extraCommonData.toMutableMap().apply {
+                putIfNotNull("accountId", accountId)
+            }
+        }
+    }
+
+    fun logPageView(pageName: String, additionalInfo: Map<String, Any>? = null) {
+        val extraInfo = additionalInfo?.toMutableMap() ?: mutableMapOf()
+        extraInfo["eventType"] = "page_view"
+        tracker?.info(pageName, extraInfo)
+    }
+
+    fun logError(eventName: String, additionalInfo: Map<String, Any>) {
+        tracker?.error(eventName, additionalInfo)
```

Comment:
The 3 methods can be merged with default values.

Reply:
Which ones? I think the parameters are different here even the function name is same.

Reply:
Oh okay. Maybe merge the 2 below then?

Reply:
Yep, fixed.